internal/sync/sync.go

```diff
@@ -178,53 +181,40 @@ func (s *Synchronizer) Status() *types.SyncStatus {
 	}
 }
 
-func (s *Synchronizer) updateState(stateDiff *types.StateDiff) error {
-	for _, deployedContract := range stateDiff.DeployedContracts {
-		err := s.SetCode(stateDiff, deployedContract)
+func (s *Synchronizer) updateState(collectedDiff *CollectorDiff) error {
+	for _, deployedContract := range collectedDiff.stateDiff.DeployedContracts {
+		err := s.SetCode(collectedDiff, &deployedContract)
 		if err != nil {
 			return err
 		}
```

Comment:
The scope can be reduced here as follows.

```suggestion
		if err := s.SetCode(collectedDiff, &deployedContract); err != nil {
			return err
		}
```

Reply:
done